CMakeLists.txt

```diff
@@ -33,34 +28,40 @@ if(EXISTS "${LOC_PATH}")
     "Please make a build subdirectory. Feel free to remove CMakeCache.txt and CMakeFiles.")
 endif()
 
-# if cuda, need to set cxx compiler to nvcc_wrapper before compilers
-# are checked
-if(SINGULARITY_USE_KOKKOS AND SINGULARITY_USE_CUDA)
-# set nvcc wrapper default compiler
-  if(NOT "$ENV{CXX}x" STREQUAL "x" AND
-     "$ENV{NVCC_WRAPPER_DEFAULT_COMPILER}x" STREQUAL "x")
-    set(ENV{NVCC_WRAPPER_DEFAULT_COMPILER} "$ENV{CXX}")
-  endif()
-  # set necessary kokkos build options if building inline
-  if(NOT SINGULARITY_KOKKOS_INSTALL_DIR)
-    set(NVCC_WRAPPER_DIR ${CMAKE_CURRENT_SOURCE_DIR}/utils/kokkos/bin CACHE STRING "")
-  else()
-    set(NVCC_WRAPPER_DIR ${SINGULARITY_KOKKOS_INSTALL_DIR}/bin CACHE STRING "")
-  endif()
-  set(CMAKE_CXX_COMPILER ${NVCC_WRAPPER_DIR}/nvcc_wrapper CACHE STRING "")
+# If the user doesn't specify a build type, prefer RelWithDebInfo
+set(default_build_type "RelWithDebInfo")
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+    STRING "Choose the type of build." FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-project(singularity-opac VERSION 0.0.1)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+add_library(${PROJECT_NAME} INTERFACE IMPORTED GLOBAL)
+# so we can use with add_subdirectory
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
-if(SINGULARITY_USE_FORTRAN)
-  enable_language(CXX Fortran)
-else()
-  enable_language(CXX)
+include (SetupDeps)
+include (SetupOptions)
+include (SetupCompilers)
+include (SetupFlags)
+
+include(GNUInstallDirs)
+include(CTest)
+
+# flag if this is the main project or being included
+set(IS_TOPLEVEL_PROJECT FALSE)
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  set(IS_TOPLEVEL_PROJECT TRUE)
 endif()
```

Comment:
Oh this is cool. I didn't know this was possible.

Reply:
Don't think this get's used (yet), but nice to have.